### PR TITLE
Implemented Product Identifier using dxp-components

### LINKED ISSUE #210: Implement product identifier using dxp-components.
## What is the motivation for adding/enhancing this feature?

Add a product identifier card in settings so that the user can change the identifier of the products displayed in app as per his/her requirement.

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by sanskar345**
### Related Issues

Closes #210

### Short Description and Why It's Useful

Implemented product Identifier using dxp-compoinents and made changes in open order, in progress, and completed views for displaying identifier.

### Screenshots of Visual Changes before/after (If There Are Any)

fulfillment

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/ionic-bopis#contribution-guideline)

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -102,6 +110,26 @@ export default defineComponent({
...
+      productIdentificationStore.$state.productIdentificationPref
+    );
+
+    // Providing productIdentificationPref to child components
+    provide('productIdentificationPref', productIdentificationPref);
+
+    // Subscribing to productIdentificationStore state change and changing value productIdentificationPref 
+    productIdentificationStore.$subscribe((mutation: any, state) => {
+        productIdentificationPref.value = state.productIdentificationPref;
+    }, {detached: true});
```
Comment (adityasharma7) [inline src/App.vue:129]: I think `detached` should not be needed

File: src/views/Login.vue [hunk 2]
```diff
@@ -74,10 +76,17 @@ export default defineComponent({
...
       if(!this.baseURL) this.store.dispatch("user/setUserInstanceUrl", this.alias[instanceURL] ? this.alias[instanceURL] : instanceURL);
       const { username, password } = this;
-      this.store.dispatch("user/login", { username, password }).then((data: any) => {
+      this.store.dispatch("user/login", { username, password }).then(async (data: any) => {
         if (data.token) {
           this.username = ''
           this.password = ''
+
+          // Get product identification from api using dxp-component and set the state if eComStore is defined
+          if(this.currentEComStore.productStoreId){
```
Comment (adityasharma7) [inline src/views/Login.vue:85]: This should be done asynchronous in login action

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 1 above)
Comment (sanskar345) [reply src/App.vue]: removed it

File: src/views/Login.vue (hunk 2 above)
Comment (sanskar345) [reply src/views/Login.vue]: done

File: src/App.vue [hunk 3]
```diff
@@ -93,6 +95,12 @@ export default defineComponent({
     if (this.userProfile && this.userProfile.userTimeZone) {
       Settings.defaultZone = this.userProfile.userTimeZone;
     }
+
+    // Get product identification from api using dxp-component and set the state if eComStore is defined
+    if (this.currentEComStore.productStoreId) {
+      await useProductIdentificationStore().getIdentificationPref(this.currentEComStore.productStoreId)
+        .catch((error) => console.log(error));
```
Comment (ymaheshwari1) [inline src/App.vue:102]: Use `logger` in place of `console`, also use error method for logging errors.

File: src/views/Settings.vue [hunk 4]
```diff
@@ -88,6 +116,9 @@
 import { 
   IonButton, 
   IonCard, 
+  IonCardHeader,
+	IonCardTitle,
```
Comment (ymaheshwari1) [inline src/views/Settings.vue:120]: Improved indentation.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 3 above)
Comment (sanskar345) [reply src/App.vue]: done

File: src/views/Settings.vue (hunk 4 above)
Comment (sanskar345) [reply src/views/Settings.vue]: done

**Comment by ymaheshwari1** [comment]
Closing, as will be handled through reusable component.

**Pull request closed without merging**